commit 38f18b370546d266738dde7491dd466515491fe6
Author: Chen-Yu Tsai <wens@csie.org>
Date:   Mon Dec 17 12:04:39 2018 +0800

    Bluetooth: hci_bcm: Handle deferred probing for the clock supply
    
    [ Upstream commit 28ac03b9ac3f784c2f048a910c8d0a7a87483b66 ]
    
    On some systems that actually have the bluetooth controller wired up
    with an extra clock signal, it's possible the bluetooth controller
    probes before the clock provider. clk_get would return a defer probe
    error, which was not handled by this driver.
    
    Handle this properly, so that these systems can work reliably.
    
    Tested-by: Ondrej Jirman <megous@megous.com>
    Signed-off-by: Chen-Yu Tsai <wens@csie.org>
    Signed-off-by: Marcel Holtmann <marcel@holtmann.org>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/bluetooth/hci_bcm.c b/drivers/bluetooth/hci_bcm.c
index ddbd8c6a0ceb..800132369134 100644
--- a/drivers/bluetooth/hci_bcm.c
+++ b/drivers/bluetooth/hci_bcm.c
@@ -907,6 +907,10 @@ static int bcm_get_resources(struct bcm_device *dev)
 
 	dev->clk = devm_clk_get(dev->dev, NULL);
 
+	/* Handle deferred probing */
+	if (dev->clk == ERR_PTR(-EPROBE_DEFER))
+		return PTR_ERR(dev->clk);
+
 	dev->device_wakeup = devm_gpiod_get_optional(dev->dev, "device-wakeup",
 						     GPIOD_OUT_LOW);
 	if (IS_ERR(dev->device_wakeup))
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1167_38f18b370546d266738dde7491dd466515491fe6.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1167_38f18b370546d266738dde7491dd466515491fe6.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
